File: 500+/530-minimum-absolute-difference-in-BST.cpp
int getMinimumDifference(TreeNode* root) {
    TreeNode* prev = nullptr;
    int diff = INT_MAX;
    inorder(root, prev, diff);
    return diff;
}

void inorder(TreeNode* root, TreeNode* &prev, int &diff) {
    if (root == nullptr) return;
    inorder(root->left, prev, diff);
    if (prev) {
        diff = min(root->val - prev->val, diff);
    }
    prev = root;
    inorder(root->right, prev, diff);
}

File: python version/530-minimum-absolute-difference-in-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        self.diff = 1000000
        self.prev = -1

        def helper(node):
            if not node: return
            helper(node.left)
            if self.prev != -1:
                self.diff = min(self.diff, node.val-self.prev)
            self.prev = node.val
            helper(node.right)

        helper(root)

        return self.diff